### PR TITLE
Profiler tasks for K-Means and DBSCAN

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp [hunk 1]
```diff
@@ -315,6 +322,7 @@ sycl::event kernels_fp<Float>::partial_reduce_centroids(
     std::int64_t part_count,
     pr::ndview<Float, 2>& partial_centroids,
     const bk::event_vector& deps) {
+    //    ONEDAL_PROFILER_TASK(partial_reduce_centroids, queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp:325]: Hang if uncommented

File: cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp [hunk 2]
```diff
@@ -225,6 +227,7 @@ std::int32_t kernels_fp<Float>::start_next_cluster(sycl::queue& queue,
                                                    const pr::ndview<std::int32_t, 1>& cores,
                                                    pr::ndview<std::int32_t, 1>& responses,
                                                    const bk::event_vector& deps) {
+    ONEDAL_PROFILER_TASK(start_nex_cluster, queue);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp:230]: `next`?

```suggestion
    ONEDAL_PROFILER_TASK(start_next_cluster, queue);
```

File: cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp [hunk 3]
```diff
@@ -322,6 +325,7 @@ sycl::event kernels_fp<Float>::update_queue(sycl::queue& queue,
                                             std::int64_t block_start,
                                             std::int64_t block_end,
                                             const bk::event_vector& deps) {
+    ONEDAL_PROFILER_TASK(update_queue, queue);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp:328]: I know that it is too late for this comment. But naming `update_queue` is confusing for me....

**Review by KulikovNikita — APPROVED** [review]

File: cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp (hunk 2 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp]: Fixed

File: cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp (hunk 3 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp]: Let's discuss it later

**Review by napetrov — APPROVED** [review]

**Review by napetrov — APPROVED** [review]

**Pull request merged**